<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>software.knit</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/ionicons-2.0.1/css/ionicons.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Bioinformatics Resource Center</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="software.html">Software</a>
</li>
<li>
  <a href="training.html">Training</a>
</li>
<li>
  <a href="analysis.html">Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RockefellerUniversity">
    <span class="ion ion-social-github"></span>
     
  </a>
</li>
<li>
  <a href="https://www.youtube.com/@brcru">
    <span class="ion ion-social-youtube"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/Rockefeller_BRC">
    <span class="ion ion-social-twitter"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">




</div>


<div id="pipelines" class="section level2">
<h2><strong>Pipelines</strong></h2>
<p>The Bioinformatics Resource Center develops, maintains and hosts a
range of pipelines and downstream analysis apps for use by The
Rockefeller University community.</p>
<p>So far these pipelines have:</p>
<ul>
<li>Processed over 25,000 samples.</li>
<li>Used in &gt; 40 publications.</li>
</ul>
<p>Important features of our pipelines include:</p>
<ul>
<li>Web based Graphical User Interfaces (GUIs) to allow users to manage
genomics data access and to process and share results from our high
throughput processing pipelines.</li>
<li>Integration with our local Genomics Resource Center.</li>
<li>Methods for inclusion of external and published datasets.</li>
<li>Robustness to machine failure/restarts</li>
<li>Portable and re-deployable software environments delivered alongside
results.</li>
<li>Highly literate and reproducible workflow documents providing
provenance over input files, code used and software versions</li>
</ul>
<p>To get learn more about our pipelines please contact us at <a
href="brc@rockefeller.edu">brc@rockefeller.edu</a> <br/><br/></p>
</div>
<div id="software" class="section level2">
<h2><strong>Software</strong></h2>
<p>The Bioinformatics Resource Center focuses on the development of
bioinformatics software for high-throughput analysis, quality control
and visualization of biological data relevant to our work at The
Rockefeller University.</p>
<p>All software developed is publicly available through our github site
or from online, peer-reviewed repositories.</p>
<p>We have developed and maintain five Bioconductor packages within the
team: profileplyr, basecallQC, ChIPQC, tracktablesm, soGGi, rfastp and
Herper. <br/><br/></p>
</div>
<div id="basecallqc" class="section level2">
<h2>basecallQC</h2>
<p>The basecallQC package provides tools to work with Illumina bcl2Fastq
(versions &gt;= 2.1.7) software.Prior to basecalling and demultiplexing
using the bcl2Fastq software, basecallQC functions allow the user to
update Illumina sample sheets from versions &lt;= 1.8.9 to &gt;= 2.1.7
standards, clean sample sheets of common problems such as invalid sample
names and IDs, create read and index basemasks and the bcl2Fastq
command.</p>
<p>Following the generation of basecalled and demultiplexed data, the
basecallQC packages allows the user to generate HTML tables, plots and a
self contained report of summary metrics from Illumina XML output
files.</p>
<p><strong>Author: Thomas Carroll and Marian Dore</strong></p>
<p><strong>Bioconductor site - <a
href="https://www.bioconductor.org/packages/release/bioc/html/basecallQC.html">Link</a></strong>
<br/><br/></p>
</div>
<div id="chipqc" class="section level2">
<h2>ChIPQC</h2>
<p>ChIPQC will automatically compute a number of quality metrics, and
provides simple ways to generate a ChIP-seq experiment quality report
which can be examined to asses the absolute and relative quality of
individual ChIP-seq samples (and their associated controls, as well as
overall quality of the experimental data.)</p>
<p><strong>Author: Thomas Carroll and Rory Stark</strong></p>
<p><strong>Bioconductor site - <a
href="https://www.bioconductor.org/packages/release/bioc/html/ChIPQC.html">Link</a></strong>
<br/><br/></p>
</div>
<div id="soggi" class="section level2">
<h2>soGGi</h2>
<p>The soGGi package provides a toolset to create genomic interval
aggregate/summary plots of signal or motif occurence from BAM and bigWig
files as well as PWM, rlelist, GRanges and GAlignments Bioconductor
objects.</p>
<p>soGGi allows for normalisation, transformation and arithmetic
operation on and between summary plot objects as well as grouping and
subsetting of plots by GRanges objects and user supplied metadata.</p>
<p>Plots are created using the GGplot2 libary to allow user defined
manipulation of the returned plot object. Coupled together, soGGi
features a broad set of methods to visualise genomics data in the
context of groups of genomic intervals such as genes, superenhancers and
transcription factor binding events.</p>
<p><strong>Author: Thomas Carroll and Gopuraja Dharmaligham</strong></p>
<p><strong>Bioconductor site - <a
href="https://www.bioconductor.org/packages/release/bioc/html/soGGi.html">Link</a></strong>
<br/><br/></p>
</div>
<div id="tracktables" class="section level2">
<h2>tracktables</h2>
<p>Visualising genomics data in genome browsers is a key step in both
quality control and the initial interrogation of any hypothesis under
investigation.</p>
<p>The organisation of large collections of genomics data (such as from
large scale high-thoughput sequencing experiments) alongside their
associated sample or experimental metadata allows for the rapid
evaluation of patterns across exper- imental groups. Broad’s
Intergrative Genome Viewer (IGV) provides a set of methods to make use
of sample metadata when visualising genomics data. As well as
identifying sample metadata within the genome browser, this sample
information can be used in IGV to group, sort and filter samples.</p>
<p>This use of sample metadata, alongside the ability to control IGV
through ports, provides the desired framework to rapidly interrogate
large cohorts of genomics data once the appropriate file structure is
built.</p>
<p>The tracktables package provides a set of tools to build IGV session
files from data-frames of sample files and their associated metadata as
well as produce IGV linked HTML reports for high-throughput
visualisation of sample data in IGV.superenhancers and transcription
factor binding events.</p>
<p><strong>Author: Thomas Carroll and Anne Pajon</strong></p>
<p><strong>Bioconductor site - <a
href="https://www.bioconductor.org/packages/release/bioc/html/trcktables.html">Link</a></strong></p>
<p><br/><br/> ## Rfastp</p>
<p>Rfastp is an R wrapper of fastp developed in c++. fastp performs
quality control for fastq files. including low quality bases trimming,
polyX trimming, adapter auto-detection and trimming, paired-end reads
merging, UMI sequence/id handling.</p>
<p>Rfastp can concatenate multiple files into one file (like shell
command cat) and accept multiple files as input.</p>
<p><strong>Author: Thomas Carroll and Wei Wang</strong></p>
<p><strong>Bioconductor site - <a
href="https://www.bioconductor.org/packages/release/bioc/html/Rfastp.html">Link</a></strong>
<br/><br/></p>
</div>
<div id="profileplyr" class="section level2">
<h2>profileplyr</h2>
<p>Quick and straightforward visualization of read signal over genomic
intervals is key for generating hypotheses from sequencing data sets
(e.g. ChIP-seq, ATAC-seq, bisulfite/methyl-seq).</p>
<p>Many tools both inside and outside of R and Bioconductor are
available to explore these types of data, and they typically start with
a bigWig or BAM file and end with some representation of the signal
(e.g. heatmap).</p>
<p>profileplyr leverages many Bioconductor tools to allow for both
flexibility and additional functionality in workflows that end with
visualization of the read signal.</p>
<p><strong>Author: Thomas Carroll and Doug Barrows</strong></p>
<p><strong>Bioconductor site - <a
href="https://www.bioconductor.org/packages/release/bioc/html/profileplyr.html">Link</a></strong>
<br/><br/></p>
</div>
<div id="herper" class="section level2">
<h2>Herper</h2>
<p>Many tools for data analysis are not available in R, but are present
in public repositories like conda. The Herper package provides a
comprehensive set of functions to interact with the conda package
managament system.</p>
<p>With Herper users can install, manage and run conda packages from the
comfort of their R session. Herper also provides an ad-hoc approach to
handling external system requirements for R packages.</p>
<p>For people developing packages with python conda dependencies we
recommend using basilisk (<a
href="https://bioconductor.org/packages/release/bioc/html/basilisk.html"
class="uri">https://bioconductor.org/packages/release/bioc/html/basilisk.html</a>)
to internally support these system requirments pre-hoc.</p>
<p><strong>Author: Thomas Carroll and Matt Paul</strong></p>
<p><strong>Bioconductor site - <a
href="https://www.bioconductor.org/packages/release/bioc/html/Herper.html">Link</a></strong>
<br/><br/></p>
</div>
<div id="clipflexr" class="section level2">
<h2>CLIPflexR</h2>
<p>CLIPflexR is a R wrapper for the CLIP Tool Kit (CTK) and additional
functions to call other external libraries into an R environment.</p>
<p>CLIPflexR makes use of the Herper library to install conda
dependencies to your machine within a conda environment using the
reticulate libraries.</p>
<p><strong>Author: Kathryn Rozen-Gagnon and Thomas Carroll</strong></p>
<p><strong>GitHub site - <a
href="https://kathrynrozengagnon.github.io/CLIPflexR/index.html">Link</a></strong></p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
